### PR TITLE
startDate / endDate constraints

### LINKED ISSUE #470: [TASK] | EndTime validation: Cannot choose time earlier than StartTime
### Description
Currently EndTime can choose anytime
### Acceptance Criteria
EndTime should only select the EndTime later than StartTime.
Two Cases: AllDay event and not AllDay event

### PR DESCRIPTION
**Pull request by candicek**
**- In the add new event page:**
Both startDate and endDate's min values are the current time. (So user can only add events for future)
Once startDate has been chosen: endDate's min value is startDate. (So the endDate can only be later than startDate)
If the user changes startDate resulting endDate is before startDate, then endDate will reset to null.
Conversion from allDay to date-time works fine.

**- In the edit event page:**
The min value of startDate is the current time. (User can only modify the startDate to be today or later than today)
The min value of endDate is the startDate. (endDate can only be later than startDate)
If the user changes startDate resulting endDate is before startDate, then endDate will reset to null.
Conversion from allDay to date-time works fine.

The above constraints apply to both allDay events and regular events.

closes #470

### CODE REVIEW

**Comment by bingzhang** [comment]
@sandeep-ps pr looks good to me. But just want to confirm that we only allow user to set startdate equal or later than today.

**Review by bingzhang — DISMISSED** [review]

**Comment by bingzhang** [comment]
hold on merge.

**Comment by sandeep-ps** [comment]
I think the user should be able to create events with a past start date. For example, if there is some event that started on a past date and continuing to a present or future date, they should be able to add after the event has started. 

The main validation needed here (also as per the issue description) is that an event start date cannot be after event end date.

File: templates/events/add-new-event.html [hunk 1]
```diff
@@ -499,18 +499,75 @@ <h5>Add Sub-Event </h5>
...
+        var yyyy = today.getFullYear();
+        if(dd<10){
+          dd='0'+dd
+        } 
+        if(mm<10){
+          mm='0'+mm
+        } 
+        date = yyyy+'-'+mm+'-'+dd;
+        var dateTime = date+'T'+'00:00:00';
+        document.getElementById("startDate").setAttribute("min", dateTime);
```
Comment (bingzhang) [inline templates/events/add-new-event.html:551]: maybe remove this line.

**Review by bingzhang — APPROVED** [review]

**Pull request merged**